ksql-engine/src/main/java/io/confluent/ksql/function/udf/string/Masker.java

```diff
@@ -74,4 +77,15 @@ private int maskCharacter(final int c) {
   static int getMaskCharacter(final String stringMask) {
     return stringMask == null ? NO_MASK : stringMask.codePointAt(0);
   }
+
+  static void validateParams(final String udfName, final int numChars) {
+    if (numChars < 0) {
+      throw new KsqlFunctionException(
+          "function " + udfName + " requires a non-negative number of characters to mask or skip");
+    }
+  }
+
+  static String getMaskUdfName(final Object maskObject) {
```

Comment:
Perhaps we should change the signature to accept a Class instead of an Object, that way this method can be called without instantiating an object of a specific type (and makes the contract a little clearer). Thoughts?

Reply:
I couldn't get the UdfDescription annotation name from the class. It had to be instantiated in order to get it. Anyway, I like the other approach you mentioned, which declares the UDF name as static (and protected), and use it in the @UdfDescription annotation. I will use that instead of calling this function.